data/templates/get_driveid.js

```
@@ -0,0 +1,196 @@
+ */
+function buildDriveMap(wwidData, vdData, scsiData) {
+    var parsedWwids = parseDriveWwid(wwidData),
+        scsiList = parseScsiInfo(scsiData),
+        vdList = parseVdInfo(vdData);
+    var linuxWwids = parsedWwids.linuxDriveIds, esxiWwids = parsedWwids.esxiDriveIds;
+    var driveIds=[];
+    esxiWwids.forEach(function(esxiWwid){
+        var vd = '', scsiId = '', diskPath = esxiWwid[0];
+        vdList.forEach(function(elem){
+            if (typeof elem !== 'undefined') {
+                if (elem[0] === diskPath) {
+                    vd = elem[1];
+                }
+            }
+        });
+        scsiList.forEach(function(elem){
+            if (typeof elem !== 'undefined'){
+                if(elem[0] === diskPath){
+                    scsiId = elem[1];
+                }
+            }
+        });
+        driveIds.push({"scsiId": scsiId, "virtualDisk": vd,
+            "esxiWwid": esxiWwid[1], "devName": diskPath});
+    });
+    var k=0;
+    linuxWwids.forEach(function(linuxWwid){
+        driveIds[k].identifier = k;
+        driveIds[k].linuxWwid = linuxWwid[1];
+        k += 1;
+    });
+    console.log(JSON.stringify(driveIds));
+    return 0;
+}
+
+function run() {
+    var wwidData, vdData, scsiData;
+    exec(cmdDriveWwid, options, function(err0, stdout0) {
+        if (err0) {
+            console.error(err0.toString());
+            process.exit(1);
+        }
+        wwidData = stdout0;
+        exec(cmdVdInfo, options, function(err1, stdout1) {
+            if (err1) {
+                console.error(err1.toString());
+                process.exit(1);
+            }
+            vdData = stdout1;
+            exec(cmdScsiId, options, function(err2, stdout2) {
+                if (err2) {
+                    console.error(err2.toString());
+                    process.exit(1);
+                }
+                scsiData = stdout2;
+                if (buildDriveMap(wwidData, vdData, scsiData)) {
```

Should this be wrapped in a try/catch to fail gracefully in case the parsing fails, or am I missing parser error handling somewhere?